File: webshop-frontend/src/services/WishListService.ts
import http from "../http/HttpCommonWishList";


class WishListService {

    getAllFromCartLoggedUser(userId: any) {
        return http.get<Array<any>>(`/findById/${userId}`);
    }

    addBikeToWishList(userId:any, bikeId: any) {
        return http.post<Array<any>>(`/${userId}/${bikeId}/bikes`);
    }

    removeBikeFromWishList(bikeId: any) {
        return http.delete<Array<any>>(`/${bikeId}/bikes`);
    }

}

export default new WishListService();

File: webshop-backend/src/main/java/com/example/webshop/service/WishListService.java
package com.example.webshop.service;

import com.example.webshop.model.WishList;

import java.util.List;

public interface WishListService {
    List<WishList> findAllById(Long userId);

    void addBikeToWishList(Long userId, Long bikeId);

    void removeBikeFromWishList(Long bikeId);

}
